gui/src/app/SPAnalysis/SPAnalysisContextProvider.tsx

```diff
@@ -35,17 +138,37 @@ const SPAnalysisContextProvider: FunctionComponent<PropsWithChildren<SPAnalysisC
         };
     }, [data])
 
+
+
+    useEffect(() => {
+        if (data != initialDataModel) return;
+
+        const query = fromSearchParams(searchParams);
+
+        // any query is set
+        if (Object.values(query).some(v => v !== null)) {
+            fetchRemoteAnalysis(query).then((data) => {
+                update({ type: 'loadInitialData', state: data })
+            })
+        } else {
+            // load the saved state on first load
+            const savedState = localStorage.getItem('stan-playground-saved-state')
+            if (!savedState) return
+            const parsedData = deserializeAnalysis(savedState)
+            update({ type: 'loadInitialData', state: parsedData })
+        }
+
+    }, [data, searchParams])
+
     useEffect(() => {
-        // load the saved state on first load
-        const savedState = localStorage.getItem('stan-playground-saved-state')
-        if (!savedState) return
-        const parsedData = deserializeAnalysis(savedState)
-        update({ type: 'loadLocalStorage', state: parsedData })
-    }, [])
+        if (searchParams.size !== 0 && modelHasUnsavedChanges(data)) {
+            setSearchParams(new URLSearchParams());
```

Comment:
Would be good to leave a comment here explaining that we are clearing the query string, and why we are doing that.